src/main/settings/settings.cpp

```diff
@@ -552,6 +552,84 @@ Value EnableProfilingSetting::GetSetting(ClientContext &context) {
 	}
 }
 
+//===--------------------------------------------------------------------===//
+// Custom Profiling Settings
+//===--------------------------------------------------------------------===//
+
+static unordered_set<TreeNodeSettingsType, TreeNodeSettingsTypeHashFunction>
+FillTreeNodeSettings(unordered_map<string, string> &json) {
+	unordered_set<TreeNodeSettingsType, TreeNodeSettingsTypeHashFunction> metrics;
+
+	string invalid_settings;
+	for (auto &entry : json) {
+		auto setting = EnumUtil::FromString<TreeNodeSettingsType>(StringUtil::Upper(entry.first));
+		if (StringUtil::Lower(entry.second) == "true") {
+			metrics.insert(setting);
+		}
+	}
+
+	if (!invalid_settings.empty()) {
+		throw IOException("Invalid custom profiler settings: \"%s\"", invalid_settings);
+	}
+	return metrics;
+}
+
+void CustomProfilingSettings::SetLocal(ClientContext &context, const Value &input) {
+	auto &config = ClientConfig::GetConfig(context);
+	auto input_str = input.ToString();
+	duckdb::unique_ptr<duckdb::FileSystem> fs = duckdb::FileSystem::CreateLocal();
+	if (!fs->FileExists(input_str)) {
+		throw IOException("Could not locate the file containing the custom profiler settings: \"%s\"", input_str);
+	}
+	if (StringUtil::GetFileExtension(input_str) != "json") {
+		throw IOException("The custom profiler settings file must be a JSON file: \"%s\"", input_str);
+	}
+	auto file = fs->OpenFile(input_str, FileFlags::FILE_FLAGS_READ);
+
+	// read file into string
+	string file_content;
+	auto line = file->ReadLine();
+	while (!line.empty()) {
+		if (StringUtil::Equals(&line.back(), "\n")) {
+			line.substr(0, line.size() - 1);
+		}
+		StringUtil::Replace(line, "\\", "");
+		file_content += line;
+		line = file->ReadLine();
+	}
+	file->Close();
+
+	// parse the file content
+	auto json = StringUtil::ParseJSONMap(file_content);
+	if (json.empty()) {
+		throw IOException("Could not parse the custom profiler settings file: \"%s\"", input_str);
+	}
+
+	auto metrics = FillTreeNodeSettings(json);
+	config.profiler_settings.SetMetrics(metrics);
+	config.profiler_settings.ResetValues();
+}
+
+void CustomProfilingSettings::ResetLocal(ClientContext &context) {
+	auto &config = ClientConfig::GetConfig(context);
+	config.profiler_settings.ResetMetrics();
+	config.profiler_settings.ResetValues();
+}
+
+Value CustomProfilingSettings::GetSetting(ClientContext &context) {
```

Comment:
Let's come back to this in a future PR, as we may want to be able to enable/disable each metric separately and only return the file path here.